pkg/webhook/policy.go

```diff
@@ -181,24 +180,34 @@ func (h *validationHandler) Handle(ctx context.Context, req admission.Request) a
 	}
 
 	res := resp.Results()
-	msgs := h.getDenyMessages(res, req)
+	msgs, enforcementAction := h.getDenyMessages(res, req)
 	if len(msgs) > 0 {
-		vResp := admission.ValidationResponse(false, strings.Join(msgs, "\n"))
+		var vResp admission.Response
 		if vResp.Result == nil {
 			vResp.Result = &metav1.Status{}
 		}
-		vResp.Result.Code = http.StatusForbidden
-		requestResponse = denyResponse
+
+		if enforcementAction == string(util.Warn) {
+			vResp = admission.ValidationResponse(true, strings.Join(msgs, "\n"))
+			vResp.Warnings = msgs
+			vResp.Result.Code = 299
+			requestResponse = allowResponse
+		} else {
+			vResp = admission.ValidationResponse(false, strings.Join(msgs, "\n"))
```

Comment:
We should add any warnings to the rejection message as warnings, so users know what's a deal breaker and what's a warning.

Reply:
Kubernetes automatically adds `Warning: ` in front of the warnings (see the example in the PR description).

```
If you're implementing a webhook that returns a warning:
Don't include a "Warning:" prefix in the message
```
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#response

Reply:
In the case where we are returning both warnings and deny messages, we need to be able to write:

```
vResp := admission.ValidationResponse(false, strings.Join(denyMsgs, "\n"))
vResp.Warnings = warnings
```